dev.c

```diff
@@ -262,17 +265,48 @@ __releases(fc->lock)
 			clear_bdi_congested(&fc->bdi, BLK_RW_ASYNC);
 		}
 		fc->num_background--;
-		fc->active_background--;
 	}
 	spin_unlock(&fc->lock);
 	if (end)
 		end(fc, req);
+
+    /* Process requests merged with this request */
+    list_for_each_entry_safe(first, next, &req->merged, merged) {
+	    first->end = NULL;
+	    first->state = FUSE_REQ_FINISHED;
+        if (end) {
+            end(fc, first);
+        }
+        fuse_put_request(fc, first);
+    }
 	fuse_put_request(fc, req);
 }
 
-static void fuse_request_send_nowait_locked(struct fuse_conn *fc,
+static bool fuse_request_send_nowait_locked(struct fuse_conn *fc,
 					    struct fuse_req *req)
 {
+    struct fuse_req *prev;
+    bool wakeup = false;
+
+    /* Check if the request could be merged with another request */
+    if (req->in.h.opcode == PXD_WRITE && !list_empty(&fc->pending)) {
+        prev = list_last_entry(&fc->pending, struct fuse_req, list);
+        if ((prev->in.h.opcode == PXD_WRITE) &&
```

Comment:
why only restrict to writes? sequential reads do help too with readaheads happening.

Reply:
I am testing changes for merging reads and ran into some issues...  This is work in progress...